src/map/pet.cpp

```diff
@@ -660,6 +660,10 @@ int pet_catch_process2(struct map_session_data* sd, int target_id)
 	//catch_target_class == PET_CATCH_UNIVERSAL is used for universal lures (except bosses for now). [Skotlex]
 	if (sd->catch_target_class == PET_CATCH_UNIVERSAL && !status_has_mode(&md->status,MD_STATUS_IMMUNE))
 		sd->catch_target_class = md->mob_id;
+	
+	//catch_target_class == PET_CATCH_UNIVERSAL_ITEM is used for catching any monster required the lure item used
+	if (sd->catch_target_class == PET_CATCH_UNIVERSAL_ITEM && sd->itemid == pet_db[i].itemID)
```

Comment:
Turn this into an else case and we are ready to merge! 👍

Reply:
you mean `else if` right ?

Reply:
Correct!